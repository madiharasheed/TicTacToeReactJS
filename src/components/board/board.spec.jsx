import React from "react";
import { shallow } from "enzyme";

import Board from "./board";
// Describing Board Components
describe("Board component", () => {
  it("should render Board component", () => {
    const mockCallBack = jest.fn();
    const boardWrapper = shallow(
      <Board tiles={Array(9).fill(null)} onClick={mockCallBack} />
    );
    expect(boardWrapper).toMatchSnapshot();
  });

  it("should have nine tiles", () => {
    const mockCallBack = jest.fn();
    const boardWrapper = shallow(
      <Board tiles={Array(9).fill(null)} onClick={mockCallBack} />
    );
    expect(boardWrapper.find("Tile").length).toEqual(9);
  });
});
