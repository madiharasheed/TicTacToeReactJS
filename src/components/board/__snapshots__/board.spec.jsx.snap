// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component should render Board component 1`] = `
<div
  className="board"
>
  <div
    className="board-row"
  >
    <Tile
      onClick={[Function]}
      value={null}
    />
    <Tile
      onClick={[Function]}
      value={null}
    />
    <Tile
      onClick={[Function]}
      value={null}
    />
  </div>
  <div
    className="board-row"
  >
    <Tile
      onClick={[Function]}
      value={null}
    />
    <Tile
      onClick={[Function]}
      value={null}
    />
    <Tile
      onClick={[Function]}
      value={null}
    />
  </div>
  <div
    className="board-row"
  >
    <Tile
      onClick={[Function]}
      value={null}
    />
    <Tile
      onClick={[Function]}
      value={null}
    />
    <Tile
      onClick={[Function]}
      value={null}
    />
  </div>
</div>
`;
